-1

Tested at: 94df57deeb9f3387aa6c8e87d0ff3632fe920fd4

CMSSW: CMSSW_11_1_X_2020-02-20-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b037ed/4808/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/MergeableCounter.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/setIsMergeable.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/WrapperBase.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/PtrVectorBase.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/RefHolder.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/RefHolder.cc:20:10: fatal error: DataFormats/Common/interfaces/RefHolderBase.h: No such file or directory
 #include <DataFormats/Common/interfaces/RefHolderBase.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/ThinnedAssociation.cc 
gmake: *** [tmp/slc7_amd64_gcc820/src/DataFormats/Common/src/DataFormatsCommon/RefHolder.cc.o] Error 1
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/MultiAssociation.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/DataFrameContainer.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/RefCoreWithIndex.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/View.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/HandleExceptionFactory.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/RefHolder.cc:20:10: fatal error: 'DataFormats/Common/interfaces/RefHolderBase.h' file not found
#include <DataFormats/Common/interfaces/RefHolderBase.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-20-2300/src/DataFormats/Common/src/RefHolder.cc:20:10: fatal error: 'DataFormats/Common/interfaces/RefHolderBase.h' file not found
#include <DataFormats/Common/interfaces/RefHolderBase.h>
</pre>

--- a/DataFormats/Common/interface/RefHolderBase.h
+++ b/DataFormats/Common/interface/RefHolderBase.h
@@ -16,7 +16,7 @@ namespace edm {
 
     class RefHolderBase {
     public:
-      RefHolderBase() {}
+      RefHolderBase();
       template <class T>
       T const* getPtr() const;
       virtual ~RefHolderBase();

--- a/DataFormats/Common/src/RefHolder.cc
+++ b/DataFormats/Common/src/RefHolder.cc
@@ -1,0 +1,8 @@
+#include "DataFormats/Common/interface/RefHolderBase.h"
+
+namespace edm {
+  namespace reftobase {
+    // Pin the vtable here.
+    RefHolderBase::RefHolderBase() {}
+  }  // namespace reftobase
+}  // namespace edm

